zio-http/src/main/scala/zio/http/api/TextCodec.scala

```diff
@@ -74,4 +80,20 @@ object TextCodec {
 
     def encode(value: UUID): String = value.toString
   }
+
+  case object EncodingCodec extends TextCodec[Encoding] {
+    override def decode(value: String): Option[Encoding] = {
+      value.split(",").flatMap(Encoding.identifyEncodingFull) match {
```

Comment:
We should avoid using `flatMap`, `split`, and other allocation-intensive methods in performance critical code. Although we can merge this in and fix later, it's best to just get into the habit of focusing on performance in the first pass for the structured header values.

Reply:
re-implemented without `split` and `flatMap`